cpp/validate_harvested_records.cc

```diff
@@ -298,12 +331,14 @@ bool RecordMeetsExpectations(const MARC::Record &record, const std::string &jour
 
 void WriteToDatabase(DbConnection * const db_connection, const GeneralInfo &general_info, const JournalInfo &journal_info) {
     for (const auto &field_info : journal_info) {
-        if (general_info.find(field_info.name_) == general_info.end())
+        if (general_info.find(field_info.name_, field_info.subfield_code_) == general_info.end())
```

Comment:
RecordType should be considered somehow here

If we have a regular_article, we only want to check against GeneralInfo for regular_article, and insert as review_article.
If we have a review here, we want to check vs regular_article as well as review, and i guess that only if both is missing we insert as review.